### ISSUE TITLE
Shift Double-click in tmux select words from differnet panes

### ISSUE THREAD

**Issue by SivanagBalla** [microsoft/terminal#17225]
### Windows Terminal version

1.20.11215.0

### Windows build number

10.0.22631

### Other Software

Tmux in WSL with multiple panes

### Steps to reproduce

Consider a tmux session with multiple panes as shown in below pic.
![image](https://github.com/microsoft/terminal/assets/19350704/73993e1c-7c8f-4858-99c3-9074b6baf462)

When i try to "shift double-click" on the word "siballa", Terminal also selects the pane delimiter line from other pane because it thinks its complete word.

Another example below where in which i tried to select the file permission string "lrwxrwxrw". It selected the "yankrink_hi" from the other pane

![image](https://github.com/microsoft/terminal/assets/19350704/bff4f34d-f5f1-4fea-b700-dd1be1df4a82)

### Expected Behavior

In tmux enviroment, Double-click should select words only from current pane

### Actual Behavior

In tmux enviroment, Double-click should select words only from multiple panes.  

This may not be bug but atleast there should be a option to prevent selection of words across multiple lines. I use tmux with multiple panes and encounter this issue often.

**Comment by carlos-zamora** [comment]
Thanks for filing! We're having trouble reproducing this. Would you be able to write up the steps required to get in this state /? This'll help us take a closer look 😊

**Comment by SivanagBalla** [comment]
@carlos-zamora Below is another screen shot where we can reproduce it - 

Open a terminal - adjust the width to say 80 or less. Width shouldn't matter here but it makes easy to repro 
Start a tmux session with two vertical split panes
Type some lengthy strings in both panes.
Now shift-doubleclick on a string on one pane - terminal will expand the selectoin to other pane.

![image](https://github.com/microsoft/terminal/assets/19350704/045ceb54-9efd-4f11-bc96-bf46ecd85549)

My word delimiters are set to  "\()"',;<>~!@#$%^&*|+[]{}~?│"

**Comment by carlos-zamora** [comment]
Does this repro on [Windows Terminal Canary](https://aka.ms/terminal-canary-installer)?

**Comment by SivanagBalla** [comment]
Yes, just download (ver:1.22.1631.0) it and i can see the issue there also

**Comment by carlos-zamora** [comment]
We think this should be fixed by a combination of #15602 and some of the recent ConPTY work we've been doing. We're going to move the info over from that issue over here though since this has a nice test for it.



From what I understand, a row should only be marked as force-wrapped if we actually wrote past the last column, which naturally occurs already in the `AdaptDispatch::_DoLineFeed` call.

Removing the `SetWrapForced` call however breaks these unit tests:
```
ConptyOutputTests::InvalidateUntilOneBeforeEnd
TerminalCoreUnitTests::ConptyRoundtripTests::BreakLinesOnCursorMovement#metadataSet0
TerminalCoreUnitTests::ConptyRoundtripTests::BreakLinesOnCursorMovement#metadataSet1
TerminalCoreUnitTests::ConptyRoundtripTests::BreakLinesOnCursorMovement#metadataSet2
TerminalCoreUnitTests::ConptyRoundtripTests::BreakLinesOnCursorMovement#metadataSet3
TerminalCoreUnitTests::ConptyRoundtripTests::BreakLinesOnCursorMovement#metadataSet4
TerminalCoreUnitTests::ConptyRoundtripTests::BreakLinesOnCursorMovement#metadataSet5
TerminalCoreUnitTests::ConptyRoundtripTests::BreakLinesOnCursorMovement#metadataSet6
TerminalCoreUnitTests::ConptyRoundtripTests::ResizeRepaintVimExeBuffer
TerminalCoreUnitTests::ConptyRoundtripTests::WrapNewLineAtBottom#metadataSet6
TerminalCoreUnitTests::ConptyRoundtripTests::WrapNewLineAtBottom#metadataSet7
TerminalCoreUnitTests::ConptyRoundtripTests::WrapNewLineAtBottom#metadataSet8
TerminalCoreUnitTests::ConptyRoundtripTests::WrapNewLineAtBottomLikeMSYS#metadataSet6
TerminalCoreUnitTests::ConptyRoundtripTests::WrapNewLineAtBottomLikeMSYS#metadataSet7
TerminalCoreUnitTests::ConptyRoundtripTests::WrapNewLineAtBottomLikeMSYS#metadataSet8
```

They break because when a `ROW` is not marked as force-wrapped the output behavior of the VtEngine and its output changes. We need to investigate why they break, what the correct behavior should be and how we can fix it. Presumably most unit tests simply assert the wrong behavior, but it's also possible that `VtEngine::_PaintUtf8BufferLine` and its `removeSpaces` variable may have to be updated.

**Comment by nanaya** [comment]
Still a problem on 1.23.250115001-llm

**Comment by DHowett** [comment]
After a lot of deliberation, we're pretty sure this is a /dup of #8976.
